HashRegistrarSimplified.sol

```diff
@@ -0,0 +1,367 @@
+    }
+
+    function max(uint a, uint b) internal constant returns (uint max) {
+        if (a > b)
+            return a;
+        else
+            return b;
+    }
+
+    function  min(uint a, uint b) internal constant returns (uint min) {
+        if (a < b)
+            return a;
+        else
+            return b;
+    }
+
+    /*
+    * @title String & slice utility library for Solidity contracts.
+    * @author Nick Johnson <arachnid@notdot.net>
+    */
+    struct slice {
+        uint _len;
+        uint _ptr;
+    }
+
+    /*
+     * @dev Returns a slice containing the entire string.
+     * @param self The string to make a slice from.
+     * @return A newly allocated slice containing the entire string.
+     */
+    function toSlice(string self) internal returns (slice) {
+        uint ptr;
+        assembly {
+            ptr := add(self, 0x20)
+        }
+        return slice(bytes(self).length, ptr);
+    }
+    
+    /*
+     * @dev Returns the length in runes of the slice. Note that this operation
+     *      takes time proportional to the length of the slice; avoid using it
+     *      in loops, and call `slice.empty()` if you only need to know whether
+     *      the slice is empty or not.
+     * @param self The slice to operate on.
+     * @return The length of the slice in runes.
+     */
+    function len(slice self) internal returns (uint) {
+        // Starting at ptr-31 means the LSB will be the byte we care about
+        var ptr = self._ptr - 31;
+        var end = ptr + self._len;
+        for (uint len = 0; ptr < end; len++) {
+            uint8 b;
+            assembly { b := and(mload(ptr), 0xFF) }
+            if (b < 0x80) {
+                ptr += 1;
+            } else if(b < 0xE0) {
+                ptr += 2;
+            } else if(b < 0xF0) {
+                ptr += 3;
+            } else if(b < 0xF8) {
+                ptr += 4;
+            } else if(b < 0xFC) {
+                ptr += 5;
+            } else {
+                ptr += 6;
+            }
+        }
+        return len;
+    }
+
+    /*
+    ## Start Auction for available hash
+
+    Anyone can start an auction by sending an array of hashes that they want to bid for. 
+    Arrays are sent so that someone can open up an auction for X dummy hashes when they 
+    are only really interested in bidding for one. This will increase the cost for an 
+    attacker from simply bidding on all new auctions blindly. Dummy auctions that are 
+    open but not bid on are closed after a week. 
+    */    
+    function startAuction(bytes32 _hash) noEther {
+        entry newAuction = entries[_hash];
+        if ((newAuction.status == Mode.Auction && now < newAuction.registrationDate)
+            || newAuction.status == Mode.Owned 
+            || newAuction.status == Mode.Forbidden
+            || now > registryCreated + 4 years)
+            throw;
+        
+        // for the first five months of the registry, make longer auctions
```

Comment:
Weren't we going to change to an extended period on launch, then transition straight to 1 week auctions?